mariamihaelafainarea/assignment3/Assignment3

```diff
@@ -93,96 +77,91 @@ public class Assignment3 {
                         }
                         //restart the state after a deep search ! 
                         start.deleteCharAt(hopa.length()-1);
-                        visited[newX][newY] = 0;
+                        visited[newX][newY] = false;
                     }
                 }
             }
         }
         
     }
+    //A dictionary based on a trie ! 
+    //It has a better complexity than
+    //a hashmap !
     class MyDictionary{
-        //A dictionary based on a trie ! 
-        //It has a better complexity than
-        //a hashmap !
         private class TrieNode{
             private HashMap<Character,TrieNode> children;
             private boolean endOfWord;
-            //endOfWord is like.. here in this node
-            //a word ends even if it is not a leaf !
             public TrieNode() {
                 children = new HashMap<>();
                 endOfWord = false;
             }
             
         }
         private final TrieNode root;
-        
+        //the idea is very simple
+        //you start from root
+        //you go deep down in the trie 
+        //with the current node
+        //if theChildren (the character that need to be added)
+        //exists you only pass it and go deep down
         public void insert(String word) {
-            //the idea is very simple
-            //you start from root
-            //you go deep down in the trie 
-            //with the current node
-            //if theChildren (the character that need to be added)
-            //exists you only pass it and go deep down
             TrieNode current = root;
             for(int i = 0; i < word.length();i++) {
-                TrieNode theChildren = current.children.get(word.charAt(i));
-                if (theChildren == null) {  
-                    theChildren = new TrieNode();
-                    current.children.put(word.charAt(i), theChildren);
+                TrieNode theChild = current.children.get(word.charAt(i));
+                if (theChild == null) {  
+                    theChild = new TrieNode();
+                    current.children.put(word.charAt(i), theChild);
                 }
-                current = theChildren;
+                current = theChild;
             }
             //after you add the last node and move your
             //current there
             //you mark it as endOfWord
             current.endOfWord = true;
             
         }
-        public boolean isWordHelper(TrieNode current,String word) {
-            if (word.equals("")) {
-                return current.endOfWord;
+        public boolean isGeneralHelper(String action,TrieNode current,String word,int index) {
+            if(action.equals("word")) {
+                if (index == word.length()) {
+                    return current.endOfWord;
+                }
+            } else {
+                if (index == word.length()) {
+                    if(current == null) {
```

Comment:
Next 4 lines can be replaced with simply return current != null;

Reply:
done